packages/validation/src/validator.ts

```diff
@@ -0,0 +1,58 @@
+import { DI } from '@aurelia/kernel';
+import { LifecycleFlags } from '@aurelia/runtime';
+import { ValidationResult, validationRulesRegistrar } from './rule-provider';
+import { ValidateInstruction } from './validation-controller';
+
+export const IValidator = DI.createInterface<IValidator>("IValidator").noDefault();
+
+/**
+ * The core validator contract.
+ */
+export interface IValidator {
+  /**
+   * Core validate function that works with a validate instruction.
+   *
+   * @param {ValidateInstruction<T>} instruction - The instruction on how to perform the validation.
+   * - case `{object}` - the default ruleset defined on the instance or the class are used.
+   * - case `{object, propertyName}` - only the rules defined for the particular property are validated.
+   * - case `{object, rules}`  or `{object, propertyName, rules}` - only the specified rules are used for validation.
+   * - case `{object, objectTag}` - only the tagged ruleset are used for validation.
+   * - case `{object, objectTag, propertyName}` - only the rules for the property in the tagged ruleset are used for validation.
+   * - case `{object, objectTag, propertyName, propertyTag}` - only the tagged rules for the property in the tagged ruleset for the object are validated
+   */
+  validate<T>(instruction: ValidateInstruction<T>): Promise<ValidationResult[]>;
+}
+
+/**
+ * Standard implementation of `IValidator`.
+ */
+export class StandardValidator implements IValidator {
+  public async validate<T>(instruction: ValidateInstruction<T>): Promise<ValidationResult[]> {
+    const object = instruction.object;
+    const propertyName = instruction.propertyName;
+    const propertyTag = instruction.propertyTag;
+
+    const validateAllProperties = propertyName === void 0;
+    const rules = instruction.rules ?? validationRulesRegistrar.get(object, instruction.objectTag) ?? [];
+
+    const result = await Promise.all(rules.reduce((acc: Promise<ValidationResult[]>[], rule) => {
+      const { name, expression } = rule.property;
+      if (validateAllProperties || name === propertyName) {
+        let value: unknown;
+        if (expression === void 0) {
+          value = object;
+        } else {
+          const scope = { bindingContext: object, parentScope: null, scopeParts: [], overrideContext: (void 0)! };
```

Comment:
I think it's something like this:

```suggestion
          const scope = Scope.create(object);
```

Might also be another overload, but there should be an api for this. `scope` really should ideally be the `Scope` class from the runtime though.

Reply:
That's nice! I have not noticed that. Thank you for pointing it out :)

Reply:
Done :)